src/reader.rs

```diff
@@ -60,6 +60,11 @@ impl<R: RawReader> Reader<R> {
         let mut is_append = false;
         let mut new_symbols = vec![];
 
+        // It's illegal for a symbol table to have multiple `symbols` or `imports` fields.
+        // Keep track of whether we've already encountered them.
+        let mut has_found_symbols_field = false;
+        let mut has_found_imports_field = false;
```

Comment:
🗺️ Here we detect and reject repeated `imports` and `symbols` in a symbol table.